llvm/lib/Target/AMDGPU/BUFInstructions.td

```diff
@@ -572,18 +572,25 @@ class MUBUF_Store_Pseudo <string opName,
   let tfe = isTFE;
 }
 
+multiclass MUBUF_Pseudo_Store_Pats<string BaseInst, ValueType store_vt = i32, SDPatternOperator st = null_frag> {
+
+  def _OFFSET : GCNPat <
+    (st store_vt:$vdata, (MUBUFOffset v4i32:$srsrc, i32:$soffset, i32:$offset)),
+    (!cast<MUBUF_Pseudo>(BaseInst # _OFFSET) store_vt:$vdata, v4i32:$srsrc, i32:$soffset, i32:$offset)>;
```

Comment:
This part of the refactor may be a slight downgrade in sense that you need to explicitly specify the operands of the instruction when the pat is separate rather than part of the instruction definition.

Reply:
This also makes it easier to fix arbitrary types for the mubuf global path